python/run_test.sh

```diff
@@ -19,8 +19,17 @@ spark-rapids-submit --master local[1] tests_no_import_change/test_no_import_chan
 # runs on cpu with spark-submit
 spark-submit --master local[1] tests_no_import_change/test_no_import_change.py 0.2
 
+# calculate pytest parallelism following https://github.com/NVIDIA/spark-rapids/blob/branch-24.12/integration_tests/run_pyspark_from_build.sh
+GPU_MEM_PARALLEL=`nvidia-smi --query-gpu=memory.free --format=csv,noheader | awk '{if (MAX < $1){ MAX = $1}} END {print int((MAX - 2 * 1024) / ((1.5 * 1024) + 750))}'`
+CPU_CORES=`nproc`
+HOST_MEM_PARALLEL=`cat /proc/meminfo | grep MemAvailable | awk '{print int($2 / (8 * 1024 * 1024))}'`
```

Comment:
Do these reflect container limits or VM limits?    And where did denominators come from?

Reply:
denominator is inherited from the script that Peixin shared. The script expects 8GB peak memory usage during the whole testing. Ours was 2.5 GB I observed. 8GB is much larger than that so also works for us. 

@pxLi Do you have an idea on whether the the /proc/meminfo reflects container limits or VM limits?

Reply:
Revised back to -n 3 instances.

Reply:
> Do you have an idea on whether the the /proc/meminfo reflects container limits or VM limits?

the info is for local machine but not the correct value in docker. please give it a hard upper bound if possible.

like 

if dynamical parallelism > upper_bound then go with upper bound